internal/resources/metal/vrf_bgp_dynamic_neighbor/resource.go

```diff
@@ -0,0 +1,174 @@
+	}
+
+	response.Diagnostics.Append(getPlanTags(ctx, plan, &createRequest.Tags)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	neighbor, _, err := client.VRFsApi.CreateBgpDynamicNeighbor(ctx, plan.GatewayID.ValueString()).
+		BgpDynamicNeighborCreateInput(createRequest).
+		Exclude(bgpNeighborExcludes).
+		Include(bgpNeighborIncludes).
+		Execute()
+
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Error creating VRF BGP dynamic neighbor range",
+			"Could not create VRF BGP dynamic neighbor range: "+err.Error(),
+		)
+	}
+
+	// Parse API response into the Terraform state
+	response.Diagnostics.Append(plan.parse(ctx, neighbor)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	// Set state to fully populated data
+	response.Diagnostics.Append(response.State.Set(ctx, &plan)...)
+}
+
+func (r *Resource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	client := r.Meta.NewMetalClientForFramework(ctx, request.ProviderMeta)
+
+	var data Model
+	response.Diagnostics.Append(request.State.Get(ctx, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	neighbor, _, err := client.VRFsApi.BgpDynamicNeighborsIdGet(ctx, data.ID.ValueString()).
+		// `created_by` is specified as a UserLimited.  To avoid an error
+		// due to missing UserLimited.id field, have to either exclude
+		// or include `created_by`
+		Exclude(bgpNeighborExcludes).
+		Include(bgpNeighborIncludes).
+		Execute()
+
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Error reading VRF BGP dynamic neighbor range",
+			"Could not read VRF BGP dynamic neighbor with ID "+data.ID.ValueString()+": "+err.Error(),
+		)
+	}
+
+	response.Diagnostics.Append(data.parse(ctx, neighbor)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	response.Diagnostics.Append(response.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// TODO: it should be possible to update tags but the API spec doesn't
+	// mention support for that.  In the meantime should changes to tags force
+	// recreating the resource?
+	var data Model
+	if diag := req.Plan.Get(ctx, &data); diag.HasError() {
+		resp.Diagnostics.Append(diag...)
+		return
+	}
+
+	if diag := resp.State.Set(ctx, &data); diag.HasError() {
+		resp.Diagnostics.Append(diag...)
+		return
+	}
+}
+
+func (r *Resource) Delete(ctx context.Context, request resource.DeleteRequest, response *resource.DeleteResponse) {
+	client := r.Meta.NewMetalClientForFramework(ctx, request.ProviderMeta)
+
+	var data Model
+	response.Diagnostics.Append(request.State.Get(ctx, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	// TODO: should we do something with the neighbor object returned here?
+	// For example: do we need to poll the API until neighbor.GetState() has
+	// as particular value?
+	_, _, err := client.VRFsApi.DeleteBgpDynamicNeighborById(ctx, data.ID.ValueString()).
```

Comment:
It looks like the status can be `deleting` but not `deleted`, which probably means we need a wait function here to poll the `BgpDynamicNeighborsIdGet` endpoint until it 404s.